test/micro/org/openjdk/bench/java/lang/invoke/MethodHandleProxiesAsIFInstanceCall.java

```diff
@@ -0,0 +1,186 @@
+@Fork(3)
+public class MethodHandleProxiesAsIFInstanceCall {
+    /**
+     * Avoids elimination of computation
+     */
+    public int i;
+
+    private static final Lookup LOOKUP = lookup();
+    private static final MethodType MT_Doable = methodType(Doable.class);
+    private static final MethodType MT_int_int = methodType(int.class, int.class);
+
+    // intentionally constant-folded
+    private static final MethodHandle constantTarget;
+    private static final Doable constantDoable;
+    private static final Doable constantHandle;
+    private static final Doable constantInterfaceInstance;
+    private static final Doable constantLambda;
+
+    // part of state object, non-constant
+    private MethodHandle target;
+    private Doable doable;
+    private Doable handle;
+    private Doable interfaceInstance;
+    private Doable lambda;
+
+    static {
+        try {
+            constantTarget = LOOKUP.findStatic(MethodHandleProxiesAsIFInstanceCall.class, "doWork", MT_int_int);
+        } catch (NoSuchMethodException | IllegalAccessException e) {
+            throw new ExceptionInInitializerError(e);
+        }
+        constantDoable = new Doable() {
+            @Override
+            public int doWork(int i) {
+                return MethodHandleProxiesAsIFInstanceCall.doWork(i);
+            }
+        };
+        constantHandle = new Doable() {
+            @Override
+            public int doWork(int i) {
+                try {
+                    return (int) constantTarget.invokeExact((int) i);
+                } catch (Error | RuntimeException e) {
+                    throw e;
+                } catch (Throwable e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        };
+        constantInterfaceInstance = MethodHandleProxies.asInterfaceInstance(Doable.class, constantTarget);
+        constantLambda = MethodHandleProxiesAsIFInstanceCall::doWork;
+    }
+
+    @Setup
+    public void setup() throws Throwable {
+        target = LOOKUP.findStatic(MethodHandleProxiesAsIFInstanceCall.class, "doWork", MT_int_int);
+        doable = new Doable() {
+            @Override
+            public int doWork(int i) {
+                return MethodHandleProxiesAsIFInstanceCall.doWork(i);
+            }
+        };
+        handle = new Doable() {
+            @Override
+            public int doWork(int i) {
+                try {
+                    return (int) target.invokeExact((int) i);
+                } catch (Error | RuntimeException e) {
+                    throw e;
+                } catch (Throwable e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        };
+        interfaceInstance = MethodHandleProxies.asInterfaceInstance(Doable.class, target);
+        lambda = (Doable) LambdaMetafactory.metafactory(LOOKUP, "doWork", MT_Doable, MT_int_int, target, MT_int_int).getTarget().invokeExact();
+    }
```

Comment:
The setup here can just re-use the constant* fields. The important part is that the benchmark methods load the value from a non-constant field (i.e. not static final).
```suggestion
    @Setup
    public void setup() throws Throwable {
        target = constantTarget;
        doable = constantDoable;
        handle = constantHandle;
        interfaceInstance = constantInterfaceInstance;
        lambda = constantLambda;
    }
```